### PR TITLE
Enable parsing of incomplete minisign keys, to enable re-indexing.

### PR DESCRIPTION
**Pull request by dlorenc**
Without this we can't properly re-run IndexKeys() on the log data because old minisign
keys were stripped to no longer contain the KeyID or Algorithm field.

Signed-off-by: Dan Lorenc <lorenc.d@gmail.com>

#### Ticket Link

Fixes

#### Release Note

```release-note

```

### CODE REVIEW

File: pkg/pki/minisign/minisign_test.go [hunk 1]
```diff
@@ -44,14 +44,33 @@ func TestReadPublicKey(t *testing.T) {
 	}
 
 	for _, tc := range tests {
-		file, err := os.Open(tc.inputFile)
-		if err != nil {
-			t.Errorf("%v: cannot open %v", tc.caseDesc, tc.inputFile)
-		}
+		t.Run(tc.caseDesc, func(t *testing.T) {
```
Comment (shibumi) [inline pkg/pki/minisign/minisign_test.go:45]: It might make sense to add more test cases in the tests slice above

Comment (shibumi) [reply pkg/pki/minisign/minisign_test.go:45]: Or do we cover all added new functionality?

Comment (dlorenc) [reply pkg/pki/minisign/minisign_test.go:45]: Reorganized the tests a bit and added one more case!

**Comment by dlorenc** [comment]
The coverage is now at 89% for that package, and 100% for the new code:

image

**Review by shibumi — APPROVED** [review]

File: cover.out [hunk 2]
```diff
@@ -0,0 +1,43 @@
+mode: set
```
Comment (bobcallaway) [inline cover.out]: did you mean to include this in your commit?

File: pkg/pki/minisign/minisign.go [hunk 3]
```diff
@@ -121,17 +121,30 @@ func NewPublicKey(r io.Reader) (*PublicKey, error) {
 
 	inputString := inputBuffer.String()
 
+	// There are three ways a minisign key can be stored.
+	// 1. The entire text key
+	// 2. A base64 encoded string
+	// 3. A legacy format we stored of just the key material (no key ID or Algorithm)
```
Comment (bobcallaway) [inline pkg/pki/minisign/minisign.go]:
```suggestion
	// 3. A legacy format we stored of just the key material (no key ID or Algorithm) due to bug fixed in https://github.com/sigstore/rekor/pull/562
```

**Review by bobcallaway — APPROVED** [review]
fix looks fine, not sure we want to merge `cover.out` file though

File: pkg/pki/minisign/minisign.go (hunk 3 above)
Comment (dlorenc) [reply pkg/pki/minisign/minisign.go]: Fixed!

File: cover.out (hunk 2 above)
Comment (dlorenc) [reply cover.out]: Whoops - left that after checking to address the coverage comments

**Review by bobcallaway — APPROVED** [review]
lgtm

**Comment by shibumi** [comment]
@bobcallaway  merging the cover.out file is not necessart if we add gocoveralls or other coverage tooling to our CI pipeline.
This should provide us with  a percentage overview for each PR. I am just not sure if its possible to show the coverage in the PR directly.

**Pull request merged**